funnel/models/user.py

```diff
@@ -27,6 +26,12 @@
 from . import BaseMixin, TSVectorType, UuidMixin, db
 from .helpers import add_search_trigger
 
+try:
+    from functools import cached_property
+except ImportError:
+    from werkzeug.utils import cached_property
```

Comment:
We can stick with werkzeug's cached_property. I checked the implementation and it is the same as Python 3.8's. The only difference is that werkzeug also includes a cache invalidation function, which Python doesn't. We need the invalidation occasionally, so it makes sense to keep importing from werkzeug.